### DISCUSSION TITLE
Midi channels None and All

### DISCUSSION THREAD

**Discussion by Wapata** [alf45tar/PedalinoMini#462] [Ideas]
Hi ! I'm back ;-)

An idea that should be a good one, can we add midi channel 0 (none) and midi channel All ?

The first one may need a explanation : 
Lets say I have 10 buttons and 1 Analog Pedal
For the buttons I want to use Control Change (or Control Change Snap)
For the pedal Repeat OverWrite

Then : I want to control 10 CCs with my pedal and only my pedal. I can't... each time I push a button, my CCs values jump and THEN I can control them with my Pedal. But if the button channel was 0 (in real, no message sent) then my CCs are directly controlled by my Pedal.

Or I don't understand a thing and we can achieve that an other way.

What do you think ?

**Comment by alf45tar** [comment]
Interesting

**Comment by Wapata** [comment]
Thank you for your feedback.

Of course the channel "all" can be usefull with PitchBend and so when the PedalinoMidi is beween a MasterMidiKeyboard and a SoundDevice.

**Comment by alf45tar** [comment]
If you want give a try to version 3.1.10 just released and let me know if does it match your expectations

**Reply by Wapata** [reply]
Nice,
I'm in travel for work, I will try as soon as I can !
This time it's a simple update with my browser, Should be easy to do :-)

**Reply by Wapata** [reply]
Well, it's not "a simple update" as I tought... Or may I miss something ?
![PedalinoMini_Installer](https://github.com/alf45tar/PedalinoMini/assets/15179434/d45f89a3-aa94-4d92-8ca5-215e355af1f9)

If i remember how I did i 4 month ago, in the end of the procedure i do have to "flash the firmware" with the https://alf45tar.github.io/PedalinoMini/installer page, the version is not a matter of compilation ? 
... Or is it ?

**Comment by Wapata** [comment]
Hi,
![PedalinoMini_firmware](https://github.com/alf45tar/PedalinoMini/assets/15179434/fd95b25a-a9c6-459f-bfa2-5dd387e4ddbc)
My board say that the firmware file is too big... Like twice as it expect.

Should i push it via an other method ? 
Thanks !

**Reply by alf45tar** [reply]
OTA update works only on board with at least 2 times the dimension of firmware of flash.
I suggest the PedalinoMini Installer via USB or BLE for public firmware and PlatformIO for sponsor firmware.

**Reply by Wapata** [reply]
Oh !
Okay, so I was misunderstanding a thing here :
https://github.com/pedalino-sponsors/PedalinoMini#how-to-build-and-upload
Once i've done that, i was thinking that the following chapter was mandatory too : 
https://github.com/pedalino-sponsors/PedalinoMini#how-to-connect-pedalinomini-to-a-wifi-network
and i've pushed again the last firmware with https://alf45tar.github.io/PedalinoMini/installer

So I will use PlatformIO  and only do the wifi settings via https://alf45tar.github.io/PedalinoMini/installer then.

**Reply by Wapata** [reply]
There is a lot of dependency failures.
I have to take care of my kid so will take a second look at that after.

**Comment by Wapata** [comment]
So !
It's compiling great but it look like that i'm as bad as it look like...
i've puted the new firmware into my PedalinoMini\firmware\ttgo-t-display folder, and opened the PedalinoMini folder with Visual Studio Code and then run the build...
But in the end the firmware still 3.1.6 and i didn't lost any configuration in my board.
I feel silly to don't know where to put the files :-/

**Comment by Wapata** [comment]
I've done "erase flash" under "Platform"
"Build" and "Upload" under General
"Upload Filesystem image" under Platform
Put back my Wifi settings under https://alf45tar.github.io/PedalinoMini/installer (pretending i've got 3.1.6 firmware)

And the the PedalinoMini is restarting when my computer try to reach it by IP.

Did all the procedure again, now it doesn't restart... but firmware didn't move. And I don't see an other midi channel.

Firmware
3.1.6
Firmware Size
1989920 bytes
Firmware Hash
13f8b4b66a245c5cfdff304e8fd34121

**Comment by Wapata** [comment]
I've got it !
By supressing nearly everything related to PedalinoMini on my computer and going back to zero i've got it.
I've seen channel "None" and "all" ! And somehow broken the Bluetooth and the config file. It's midnight soon so, let's sleep and see that tomorrow !

Thank you for your patience.

**Comment by Wapata** [comment]
Hi !
I'm back and finally it doesn't work as expected for the "midi all".
I suspect the "midi all" or "midi omni" is a special message, and that PedalinoMini is sending 16 messages for a command. I'm not able to make "midi learn" on other devices with Pedalino MidiAll because of this.... I guess.

(and this firmware version is much more instable, it's hard to reach the wifi-pages without many many devices reboot)

Have a nice day !
